### ISSUE TITLE
Feature request: lift derived instances `via Barbie`

### ISSUE THREAD

**Issue by gergoerdi** [fumieval/barbies-th#1]
Currently, if I want to use Barbies-TH to create a HKD that has per-field `Semigroup` and `Monoid` instances, I can do it like so:

```
declareBareB [d|
  data CPUOut = CPUOut
      { progAddr :: PC
      , memAddr :: Ptr
      , memWrite :: Maybe Cell
      , output :: Maybe Cell
      , inputNeeded :: Bool
      }|]
deriving via Barbie (CPUOut Covered) f instance (forall a. Semigroup (f a)) => Semigroup (CPUOut Covered f)
deriving via Barbie (CPUOut Covered) f instance (forall a. Monoid (f a)) => Monoid (CPUOut Covered f)
```

Would it be possible to automate this / hide the `deriving via` magic, and allow the user to instead write

```
declareBareB [d|
  data CPUOut = CPUOut
      { progAddr :: PC
      , memAddr :: Ptr
      , memWrite :: Maybe Cell
      , output :: Maybe Cell
      , inputNeeded :: Bool
      }
      deriving (Semigroup, Monoid)|]
```

which would generate the same code as above?

**Comment by fumieval** [comment]
Seems like a good idea. I'll give it a try

**Comment by fumieval** [comment]
Implemented and upload 0.1.1; it should work for most stock classes, Semigroup and Monoid

**Comment by gergoerdi** [comment]
Thanks!
